src/odemis/driver/tescan.py

```diff
@@ -128,6 +128,12 @@ def __init__(
         """
         self.device = device
         self.device_type = device_type
+        # Use a shared lock attached to the underlying SharkSEM device
+        # so multiple DeviceHandler instances using the same device work nicely together.
+        if not hasattr(self.device, "_sharksem_lock"):
+            # attach a lock to the device object
+            self.device._sharksem_lock = threading.Lock()
```

Comment:
This feels very much like the DeviceHandler should represent a connection. I think you could move the connect_device function into this class, and then all calls to the SEM go through a DeviceHandler. It might require an new `_disconnect()` function too, or maybe it can be done just automatically with a `__del__`. 

Or maybe plan this for later as a "Tech Debt"?